implementations/micrometer-registry-statsd/src/main/java/io/micrometer/statsd/StatsdMeterRegistry.java

```diff
@@ -261,7 +255,9 @@ private void startPolling() {
 
     public void stop() {
         if (started.compareAndSet(true, false)) {
-            client.dispose();
+            if (client.get() != null) {
+                client.get().dispose();
```

Comment:
For some reason I don't understand, the test for stopping and starting the meter registry did not pass when I called `dispose` on the `client` (`Disposable.Swap`) directly. If someone (maybe @simonbasle) can help me understand, it would be appreciated.

Reply:
calling `Swap.dispose()` prevents further reuse of the `Swap` container. It is basically marking the `Swap` as "I will never reuse this again, please immediately dispose its contents and every `Disposable` I attempt to set into it".

What you could do is `client.update(null)` to achieve the same. or maybe more readable: `client.update(Disposables.disposed())` (when you stop you dispose the content of the swap by updating it to be a singleton disposed instance)